code/game/mecha/mecha_decal_kits.dm

```diff
@@ -0,0 +1,561 @@
+// Paint guns accept pattern chips (/obj/item/mecha_decal_container) which can be loaded into them by attacking the gun with the chip, similar to a magazine.
+//
+// Each pattern chip grants access to one or more decals (/datum/mecha/mecha_decal) which can then be placed onto a compatible mech as a group. It is possible to create a custom paint job 
+// and decal conmbination, ejecting it from the paint gun as a pattern chip. This chip can then be placed in other paint guns or back into the current one to create a custom pattern.
+// 
+// Standard paint guns also include a nanotrasen logo already in the memory banks. Hail corporate!
+//
+
+/obj/item/mecha_paint_gun
+    name = "exosuit paint gun"
+    desc = "An automated gun-shaped device used to apply or strip off paint and other decals, compatible with most standard exosuit models."
+    icon = 'icons/mecha/mecha_decals.dmi'
+    icon_state = "mech-painter"
+
+    var/emagged = FALSE
+    var/install_sound = 'sound/items/deconstruct.ogg'
+    var/spray_sound = 'sound/effects/spray2.ogg'
+    var/icon/overlay_icon
+
+    w_class = WEIGHT_CLASS_SMALL
+    slot_flags = SLOT_BELT
+
+    // Decal kits must be installed in the exosuit paint gun prior to use.
+    var/list/obj/item/mecha_decal_container/available_decal_containers
+
+    var/obj/mecha/current_mecha
+    var/next_pattern_at = 0     // Limits the rate at which the user can generate pattern chips.
+
+/obj/item/mecha_paint_gun/Initialize()
+    available_decal_containers = list(new/obj/item/mecha_decal_container/nanotrasen_logo)
+    colour_overlay("#000000")
+    . = ..()
+
+/obj/item/mecha_paint_gun/proc/colour_overlay(var/colour)
+    overlays.Cut()
+    overlay_icon = new(icon, "mech-painter-overlay")
+    overlay_icon.Blend(colour, ICON_ADD)
+    overlays += overlay_icon
+
+/obj/item/mecha_paint_gun/attack_obj(obj/O, mob/user)
+    if(istype(O, /obj/mecha))
+        openUI(O, user)
+    else
+        . = ..()
+
+/obj/item/mecha_paint_gun/attackby(obj/O, mob/user)
+    if(istype(O, /obj/item/mecha_decal_container))
+        install_decal_container(O)
+        user.drop_item()
+        O.loc = null
+    else
+        . = ..()
+
+/obj/item/mecha_paint_gun/proc/install_decal_container(var/obj/item/mecha_decal_container/D)
+    for(var/obj/item/mecha_decal_container/C in available_decal_containers)
+        if(C.decal_name == D.decal_name)
+            to_chat(usr, "<span class='warning'>There is already a chip of that type installed!</span>")
+            return
+    available_decal_containers.Add(D)
+    to_chat(usr, "<span class='notice'>You slot [D] into [src].</span>")
+    playsound(loc, install_sound, 50, 1)
+
+/obj/item/mecha_paint_gun/proc/delete_decal_container(var/obj/item/mecha_decal_container/D)
+    available_decal_containers.Remove(D)
+
+/obj/item/mecha_paint_gun/proc/UI_colour_box(var/colour)
```

Comment:
var implied last one I'll post. Tons of `var`s in the parameters. It's not required and standard to leave out